internal/crypt/crypt_test.go

```diff
@@ -41,3 +42,9 @@ func Test_crypt_PasswordIsCrypted(t *testing.T) {
 		})
 	}
 }
+
+func TestCryptSHA512(t *testing.T) {
+	retPass, err := CryptSHA512("testPass")
+	assert.NoError(t, err)
+	assert.True(t, PasswordIsCrypted(retPass))
```

Comment:
This last line is mixing the scenarios for the 2 functions a bit. You are using the PasswordIsCrypted function to validate the result of CrytpSHA512

- For `CryptSHA512` you really don't have much to validate b/c it is a very straight forward function. The resulting string should start with `$6$` and nothing else. There should be no error.

- for `genSalt` - the behavior is that it always returns random strings and there is no error. 
- `PasswordIsCrypted` - the behavior is that it recognizes an encrypted password with one of several algorithms.

Reply:
Thanks for breaking it down like this. 

> This last line is mixing the scenarios for the 2 functions a bit. You are using the PasswordIsCrypted function to validate the result of CrytpSHA512
> 
I thought it would be easy to use that to verify it actually returns an encrypted pass. So it's preferable to verify the output without calling another function from the same `.go` file? I'm not actually trying to test the `PasswordIsCrypted` by that...

For `genSalt` I'm not sure what's the best way to verify that it always returns a random string. Should I just call it ~10 times and see if all the values are different?

Reply:
> I thought it would be easy to use that to verify it actually returns an encrypted pass. So it's preferable to verify the output without calling another function from the same `.go` file? I'm not actually trying to test the `PasswordIsCrypted` by that...

At least I always prefer to have one test case per function. Sometimes you need to use other functions in order to set-up/tear-down/verify the test case, but at least I like to make it clear which of the functions you are testing in each case, and not mix several into one.

> For `genSalt` I'm not sure what's the best way to verify that it always returns a random string. Should I just call it ~10 times and see if all the values are different?

I'd say verifying the format (length of strings, possibly characters used) and then that two calls give different values (I think you don't get much more value from >2).

Reply:
Also need to point out that if `PasswordIsCrypted` has a bug that will also break the SHA512 test so I prefer for the asserts to be as independent as possible.

Reply:
@atodorov @teg Thanks for the advice and explanation, makes sense to me. I made the changes in the latest commit and also added a separate test function for `genSalt`.